### PR TITLE
Feature/proj offset

### PR DESCRIPTION
**Pull request by thomassedlmayer**
This PR adds the PROJ offset mechanism as already defined in OpenDrive. See [OpenDrive Specification](https://publications.pages.asam.net/standards/ASAM_OpenDRIVE/ASAM_OpenDRIVE_Specification/latest/specification/08_coordinate_systems/08_05_geo_referencing.html).

### CODE REVIEW

File: doc/architecture/reference_points_coordinate_systems.adoc [hunk 1]
```diff
@@ -12,7 +12,7 @@ Coordinate system for all entities that are part of ground truth.
 The global coordinate system is an inertial x/y/z-coordinate system.
 The origin is the global reference point that is determined by the environment simulation.
 This reference point may be derived from map data or other considerations.
-Global coordinates can be mapped to a geographic coordinate system via `osi3::GroundTruth::proj_string`.
+Global coordinates can be mapped to a geographic coordinate system via `osi3::GroundTruth::proj_string`. Note that before applying any PROJ transformations to global coordinates, the `osi3::GroundTruth::proj_frame_offset` must be applied.
```
Comment (pmai) [inline doc/architecture/reference_points_coordinate_systems.adoc:15]:
```suggestion
Global coordinates can be mapped to a geographic coordinate system via `osi3::GroundTruth::proj_string`.
Note that before applying any PROJ transformations to global coordinates, the `osi3::GroundTruth::proj_frame_offset` must be applied.
```

One sentence per line

**Review by pmai — APPROVED** [review]
Looks good to me, minor formatting fix applied, don't know about the complexity of the yaw but have no opinion on it either.

**Review by thempen — COMMENTED** [review]
I Think this proposal is very good. I haven't seen any direct mistakes. As my only remark: I stumbled over the "xOSI" vs the "xWorld" definition. We have many definition of data in world coordinates within the whole OSI standard. Shouldn't they be renamed to xOSI then? Otherwise the formula should be "xWorld = xWorld * cos(yaw) - xWorld * sin(yaw) + xOffset", where xWorld will be with the ProjFramOffset, if one is available. All this naturally applies to the y coordinate as well.

**Review by thempen — APPROVED** [review]
The PR is clearly understandable and good. 
The Workgroup Harmonization & Other Models approves the PR and suggests, that the usage of "World" in other OSI messages shoud be reworked and clarified, if neccessary.

**Comment by pmai** [comment]
CCB 2024-03-25: Merge as-is, handle world coordinate system confusions in separate PR for issue #796.

**Comment by jdsika** [comment]
Reviewed for v3.7.0

**Pull request merged**